src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -379,6 +391,240 @@ void CascadiaSettings::_LoadDynamicProfiles()
+        const auto source = protoExtFolder.path().filename().wstring();
+
+        // Only apply the stubs if the parent folder name is not in ignored namespaces
+        // (also make sure this is a directory for sanity)
+        if (std::filesystem::is_directory(protoExtFolder) && ignoredNamespaces.find(source) == ignoredNamespaces.end())
+        {
+            const auto jsonFiles = _AccumulateJsonFilesInDirectory(protoExtFolder.path().c_str());
+            _AddOrModifyProfiles(jsonFiles, winrt::hstring{ source });
+        }
+    }
+}
+
+// Method Description:
+// - Finds all the json files within the given directory
+// Arguments:
+// - directory: the directory to search
+// Return Value:
+// - A set containing all the found file data
+std::unordered_set<std::string> CascadiaSettings::_AccumulateJsonFilesInDirectory(const std::wstring_view directory)
+{
+    std::unordered_set<std::string> jsonFiles;
+
+    // Expand out environment strings like %LOCALAPPDATA% and %ProgramData%
+    const std::filesystem::path root{ wil::ExpandEnvironmentStringsW<std::wstring>(directory.data()) };
+
+    for (const auto& protoExt : std::filesystem::directory_iterator(root))
+    {
+        if (protoExt.path().extension() == jsonExtension)
+        {
+            wil::unique_hfile hFile{ CreateFileW(protoExt.path().c_str(),
+                                                 GENERIC_READ,
+                                                 FILE_SHARE_READ | FILE_SHARE_WRITE,
+                                                 nullptr,
+                                                 OPEN_EXISTING,
+                                                 FILE_ATTRIBUTE_NORMAL,
+                                                 nullptr) };
+            if (!hFile)
+            {
+                THROW_LAST_ERROR();
+            }
+            else
+            {
+                const auto fileData = _ReadFile(hFile.get()).value();
+                jsonFiles.emplace(fileData);
+            }
+        }
+    }
+    return jsonFiles;
+}
+
+// Method Description:
+// - Given a set of json files, uses them to modify existing profiles,
+//   create new profiles, and create new color schemes
+// Arguments:
+// - files: the set of json files (each item in the set is the file data)
+// - source: the location the files came from
+void CascadiaSettings::_AddOrModifyProfiles(const std::unordered_set<std::string> files, const winrt::hstring source)
+{
+    for (const auto& file : files)
+    {
+        // A file could have many new profiles/many profiles it wants to modify/many new color schemes
+        // so we first parse the entire file into one json object
+        Json::Value fullFile;
+
+        _ParseJsonInto(file.data(), fullFile);
+
+        if (fullFile.isMember(JsonKey(ProfilesKey)))
+        {
+            // Now we separately get each stub that modifies/adds a profile
+            for (auto& profileStub : fullFile[JsonKey(ProfilesKey)])

```

Comment:
We add a `guid` to the stub on line 573 so that we can find a matching profile. I suppose its technically not needed and we could manually search the profile list here for a profile with the same guid as this stub's `updates` guid?

Reply:
oh, i see. eh. keep it, comment it.